### PR TITLE
MINOR: made ui enhancement in setting page

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes:

Issue #15305

- fix the user page layout when no data is there
- added button loader in test email button unless fetch is completed.

Issue : 

image

https://github.com/open-metadata/OpenMetadata/assets/66266464/66f055d9-68b1-497e-af0e-a29b3147f0ff

Resolved : 

image

https://github.com/open-metadata/OpenMetadata/assets/66266464/91e392c0-8159-478d-86b6-b79989803531

#
### Type of change:

- [x] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/DataQuality/AddDataQualityTest/TestSuiteIngestion.tsx [hunk 1]
```diff
@@ -184,11 +185,13 @@ const TestSuiteIngestion: React.FC<TestSuiteIngestionProps> = ({
 
   const handleIngestionSubmit = useCallback(
     (repeatFrequency: string) => {
+      setIsLoading(true);
       if (ingestionFQN) {
         onUpdateIngestionPipeline(repeatFrequency);
       } else {
         createIngestionPipeline(repeatFrequency);
       }
+      setIsLoading(false);
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/DataQuality/AddDataQualityTest/TestSuiteIngestion.tsx:194]: don't we need async await here?

File: openmetadata-ui/src/main/resources/ui/src/components/Settings/Email/TestEmail/TestEmail.component.tsx [hunk 2]
```diff
@@ -25,15 +25,18 @@ interface TesEmailProps {
...
   const handleSubmit: FormProps['onFinish'] = async (values) => {
     try {
+      setIsLoading(true);
       const res = await testEmailConnection(values);
       showSuccessToast(res.data);
     } catch (error) {
       showErrorToast(error as AxiosError);
     } finally {
       onCancel();
+      setIsLoading(false);
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/Settings/Email/TestEmail/TestEmail.component.tsx:39]:
```suggestion
      setIsLoading(false);
      onCancel();
```

**Review by ShaileshParmar11 — COMMENTED** [review]
few comments

File: openmetadata-ui/src/main/resources/ui/src/components/Settings/Applications/AppDetails/AppDetails.component.tsx [hunk 3]
```diff
@@ -376,7 +399,13 @@ const AppDetails = () => {
           ]
         : []),
     ];
-  }, [appData, jsonSchema]);
+  }, [
+    appData,
+    jsonSchema,
+    loadingState.isRunLoading,
+    loadingState.isSaveLoading,
+    loadingState.isDeployLoading,
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Settings/Applications/AppDetails/AppDetails.component.tsx:407]:
```suggestion
    loadingState
```

File: openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts [hunk 4]
```diff
@@ -245,7 +245,7 @@ export const updateUserAccessToken = async ({
 };
 
 export const revokeAccessToken = async (params: string) => {
-  const response = await APIClient.put<PersonalAccessToken>(
+  const response = await APIClient.put<PersonalAccessToken[]>(
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts:248]: @Ashish8689 , why did we change PersonalAccessToken to PersonalAccessToken[]?

File: openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts [hunk 5]
```diff
@@ -235,7 +235,7 @@ export const updateUserAccessToken = async ({
       JWTTokenExpiry?: string;
       tokenName?: string;
     },
-    AxiosResponse<PersonalAccessToken[]>
+    AxiosResponse<PersonalAccessToken>
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts:238]: @Ashish8689 , can you please explain this change?

Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/rest/userAPI.ts:238]: so create `personalAccessToken` for `PUT` request returns us the `PersonalAccessToken ` and not the array of it.
image
image

**Comment by Ashish8689** [comment]
@Sachin-chaurasiya @ShaileshParmar11 changes made as per comments!

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**